test/auth-route-test.js

```diff
@@ -0,0 +1,97 @@
+'use strict';
+
+const expect = require('chai').expect;
+const request = require('superagent');
+const mongoose = require('mongoose');
+const Promise = require('bluebird');
+const User = require('../model/user.js');
+
+mongoose.Promise = Promise;
+
+require('../server.js');
+
+const url = `http://localhost:${process.env.PORT}`;
+
+const exampleUser = {
+  username: 'exampleuser',
+  password: '1234',
+  email: 'exampleuser@test.com'
+};
+
+describe('Auth Routes', function() {
+  describe('POST: /api/signup', function() {
+    describe('with a valid body', function() {
+      after( done => {
+        User.remove({})
+        .then( () => done())
+        .catch(done);
+      });
+
+      it('should return a token', done => {
+        request.post(`${url}/api/signup`)
+        .send(exampleUser)
+        .end((err, res) => {
+          if (err) return done(err);
+          console.log('\ntoken:', res.text, '\n');
+          expect(res.status).to.equal(200);
+          expect(res.text).to.be.a('string');
+          done();
+        });
+      });
+    });
+  });
+
+  describe('with an invalid body', function() {
+    it('should return a 400 for a bad request', (done) => {
+      request.post(`${url}/api/signup`)
+      .send('nothing here')
+      .set('Content-Type', 'application/json')
+      .end((err, res) => {
+        expect(res.status).to.equal(400);
+        done();
+      });
+    });
+  });
+
+  describe('GET: /api/signin', function() {
+    describe('with a valid body', function() {
+      before( done => {
+        let user = new User(exampleUser);
+        user.generatePasswordHash(exampleUser.password)
+        .then( user => user.save())
+        .then( user => {
+          this.tempUser = user;
+          done();
+        })
+        .catch(done);
+      });
+
+      after( done => {
+        User.remove({})
+        .then( () => done())
+        .catch(done);
+      });
+
+      it('should return a token', done => {
+        request.get(`${url}/api/signin`)
+        .auth('exampleuser', '1234')
+        .end((err, res) => {
+          if (err) return done(err);
+          expect(res.status).to.equal(200);
+          done();
+        });
+      });
+
+      describe('without a valid auth', function() {
+        it('should return a 401 error', done => {
+          request.get(`${url}/api/signin`)
+          .auth('test user', 'wrong pwrd')
```

Comment:
this test is not returning 401 because it's actually being caught earlier by a different error. Because you are passing both an incorrect username and an incorrect password mongoose is actually going to catch the incorrect username first. It does this as part of the `User.findOne()` in line 30 of the auth-router.js.

To make this test you need to pass a correct username and incorrect password:
``` javascript
describe('without a valid auth', function() {
        it('should return a 401 error', done => {
          request.get(`${url}/api/signin`)
          .auth('testuser', 'wrong pwrd')
          .end((err, res) => {
            expect(res.status).to.equal(401);
            done();
          });
        });
      });
```